xapian-core/api/valuerangeproc.cc

```diff
@@ -439,4 +439,100 @@ NumberRangeProcessor::operator()(const string& b, const string& e)
     return Xapian::Query(Xapian::Query::OP_INVALID);
 }
 
+static const int size_map[26] = {
+    0,
+    1, // B
+    0, 0, 0, 0,
+    1024 * 1024 * 1024, // G
+    0, 0, 0,
+    1024, // K
+    0,
+    1024 * 1024, // M
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+};
+
+static int
+get_bytes(char ch)
+{
+	// It takes character as input and return number of bytes
+	// B - 1 byte
+	// K - 1024 bytes
+	// M - 1024*1024 bytes
+	// G - 1024*1024*1024 bytes
+	if (ch >= 'A' && ch <= 'Z' && size_map[ch - 'A'] > 0) {
+    return size_map[ch - 'A'];
+	} else {
+	return 0;
+	}
+}
+
+Xapian::Query
+FileSizeRangeProcessor::operator()(const string& b, const string& e){
+	// Here b and e will be like "100K" and "1M"
+	double size_b, size_e;
+	char unit_b, unit_e;
+	string temp_b = b, temp_e = e;
+
+	if (!b.empty()) {
+		errno = 0;
+		char b_back = b.back();
+		if (b_back =='B' || b_back =='K' || b_back =='M' || b_back =='G') {
```

Comment:
You should be able to do this without having the list of allowed suffices twice — once here and once, effectively, in the data structure that drives `get_bytes()`.